profiles/car.lua

```diff
@@ -390,6 +392,78 @@ function process_way(profile, way, result)
   }
 
   WayHandlers.run(profile,way,result,data,handlers)
+
+  -- now process relations data
+  local matched_refs = nil;
+  if result.ref then
+    local match_res = Relations.match_to_ref(relations, result.ref)
+
+    order = match_res['order']
+    matched_refs = match_res['match']
+
+    local ref = ''
+    for _, k in pairs(order) do
+      local v = matched_refs[k]
+      if ref ~= '' then
+        ref = ref .. '; '
+      end
+
+      if v then
+        ref = ref .. k .. ' $' .. v
+      else
+        ref = ref .. k
+      end
+    end
+
+    result.ref = ref
+  end
+
+end
+
+function process_relation(profile, relation, result)
+  local t = relation:get_value_by_key("type")
+
+  function add_extra_data(m)
+    local name = relation:get_value_by_key("name")
+    if name then
+      result[m]["route_name"] = name
+    end
+
+    local ref = relation:get_value_by_key("ref")
+    if ref then
+      result[m]["route_ref"] = ref
```

Comment:
[`route_ref`](https://taginfo.openstreetmap.org/keys/route_ref) happens to be quite common in OpenStreetMap already. Is it safe to overload this tag? If not, perhaps we could prefix it, like `osrm:route_ref`.

Reply:
This code just add data with key `reoute_ref` into relation data, that would be passed into process_way function. So it can by any we want, because it doesn't intersect or merge with OSM tags. This are two separated structures (tags, and relations data)